crates/libs/bindgen/src/extensions/variant_bool.rs

```diff
@@ -0,0 +1,75 @@
+use super::*;
+
+pub fn gen() -> TokenStream {
+    quote! {
+        impl VARIANT_BOOL {
+            #[inline]
+            pub fn as_bool(self) -> bool {
+                debug_assert!(self == VARIANT_TRUE || self == VARIANT_FALSE, "{:?} is in an invalid state", self);
+                self == VARIANT_TRUE
```

Comment:
```suggestion
                self.0 != 0
```

I think we should avoid assuming too much about how VARIANT_BOOL will be used in practice. Since its a 16-bit value I'm sure there is some ancient piece of code somewhere that breaks this assumed invariant, much like `BOOL`.